### DISCUSSION TITLE
Client upstream settings are ignored on DNS-over-TLS requests.

### DISCUSSION THREAD

**Discussion by moridew** [AdguardTeam/AdGuardHome#4812] [Q&A] [answered]
Hello.
Thank you for providing good software.

Internal DNS is being operated separately as needed.
In the client settings, the 10.9.8.0/24 range was set to "내부IP".
In the "내부IP" client, the "morid.......e domain" was set to 127.0.0.1:57053 upstream.
![client_setting](https://user-images.githubusercontent.com/83013847/183098657-5b7f1f8b-68ff-4160-b5ee-79c7fa6fb059.jpg)

Now, in plaintext DNS, it works fine
When it is DNS-over-TLS, it responds by asking the global upstream to ignore this setting.
![tls_no_change_upstream](https://user-images.githubusercontent.com/83013847/183098969-80b52e4a-7aef-4827-a0aa-f62f3c858fec.jpg)

Am I doing something wrong?
Please help me
.

**Comment by ainar-g** [comment, marked as answer]
Hello.  I cannot reproduce this.  Are you using a TLS domain name with a ClientID?

**Reply by moridew** [reply]
thank you reply.

Yes, that's right. Using clientId. 
When using clientid, is it possible to query internal dns only on specific ip?

**Reply by ainar-g** [reply]
You should either add that ClientID to the client (by pressing the :heavy_plus_sign: button and pasting the ClientID there) or remove it from the hostname that devices use so that AGH could detect the client by the IP subnet.

**Reply by moridew** [reply]
Thank you, if you remove the clientID from the DoT, it will act as a successfully configured subnet. However, I would like to have a way to operate subnet classification even when the clientID is set. Because DoT-enabled devices are Android devices, but only internal DNS servers can be used when accessing the internal network

**Reply by moridew** [reply]
I achieved what I wanted.

The reason why I'm going to do this is,
When looking up a specific domain on the internal network, it was to respond to the internal IP.
(If you receive an external IP response from the internal network and connect to the internal server via NAT Loopback, the internal server logs the internal IP of the router.)

To do this, I set up the client and let the specific domain use a separate DNS inside. However, if clientID is set in DoT or DoH, the client setting is not applied.

I decided to use Custom filtering rules.

||morid..........................e^$dnsrewrite=10.9.8.141,client='10.9.8.0/24'

When I apply the above rule, it finally works as I want.

On Android devices with DoT settings,
When it is an external network, IP is answered through cloudflare or Google DNS
On the internal network, 10.9.8.141 is answered through the dnsrewite rule.

Thank you. Have a nice day!
